### PR TITLE
added role entity and roleRepositiry, implemented springSecurity and used PreAuthorized anotation in bookController

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

**Review by annton-shakur — APPROVED** [review]
Great job!

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+                    foreignKeyName: fk_users_roles_user_id
+                    nullable: false
+                    references: users(id)
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    foreignKeyName: fk_users_roles_role_id
+                    nullable: false
+                    references: roles(id)
```
Comment (bhdnchui) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:22]: Add an empty line

Comment (bhdnchui) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:22]: Add
```
                    deleteCascade: true
                    primaryKey: true
```

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Great job! Just one point to consider.

**Review by bhdnchui — APPROVED** [review]
LGTM

**Pull request merged**